src/main/java/com/example/bank/userRegister/UserService.java

```diff
@@ -0,0 +1,60 @@
+package com.example.bank.userRegister;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.time.Period;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class UserService {
+    private final UserRepository userRepository;
+
+    User registerUser(UserRequest userRequest) {
+        log.info("Start user registration.");
+        int years = Period.between(userRequest.getBirthDate(), LocalDate.now())
+                .getYears();
+        if (years >= 18) {  //komentarz dla mnie-doczytaj o walidacji
+            User user = User.builder()
+                    .firstName(userRequest.getFirstName())
+                    .lastName(userRequest.getLastName())
+                    .birthDate(userRequest.getBirthDate())
+                    .email(userRequest.getEmail())
+                    .address(userRequest.getAddress())
+                    .build();
+            userRepository.save(user);
+            return user;
+        } else {
+            throw new IllegalArgumentException("User have to be adult.");
+        }
+    }
+
+    User getUser(Long id) throws UserNotFoundException {
+        return userRepository.findById(id)
```

Comment:
sprawdz metode getById z repository powinna byc domyslnie zrobiona przez JPA i ona zwraca optional. to juz raczej tak lepiej to obsluzyc niz w ten sposob

Reply:
Co do <T> getById(ID id) z JPA to w komentarzu dodałem, że jest deprecated i zwraca generyka, a nie Optionala. 
Zmiany zrobione + mln nowych problemów. Jeżeli możesz jakoś 'zaakceptować' plik yml, to byłoby super. 
Nie bardzo wiem jak zrobić PR z nim, bo commitowałem go z developa.

Reply:
`Optional<T> findById(ID id);
`

Reply:
Tak, jest zastosowana metoda Optional<T> findById(ID id);